nuxeo-features/rest-api/nuxeo-rest-api-server/src/main/java/org/nuxeo/ecm/restapi/server/jaxrs/usermanager/AbstractUMObject.java

```diff
@@ -58,9 +58,12 @@ public T doGetArtifact() {
     }
 
     @PUT
-    public T doUpdateArtifact(T principal) {
+    public T doUpdateArtifact(T artifact) {
+        // check the current artifact can be updated
         checkUpdateGuardPreconditions();
-        return updateArtifact(principal);
+        // check the new artifact can be updated
+        checkUpdateGuardPreconditions(artifact);
```

Comment:
check the new artifact can be updated ->  check **that** the new artifact can be updated